srtcore/core.cpp

```diff
@@ -9496,36 +9311,32 @@ void CUDT::checkTimers()
 {
     // update CC parameters
     updateCC(TEV_CHECKTIMER, TEV_CHT_INIT);
-    // uint64_t minint = (uint64_t)(m_ullCPUFrequency * m_pSndTimeWindow->getMinPktSndInt() * 0.9);
-    // if (m_ullInterval_tk < minint)
-    //   m_ullInterval_tk = minint;
-    // NOTE: This commented-out ^^^ code was commented out in original UDT. Leaving for historical reasons
 
-    uint64_t currtime_tk;
-    CTimer::rdtsc(currtime_tk);
+    const steady_clock::time_point currtime = steady_clock::now();
 
     // This is a very heavy log, unblock only for temporary debugging!
 #if 0
-    HLOGC(mglog.Debug, log << CONID() << "checkTimers: nextacktime=" << FormatTime(m_ullNextACKTime_tk)
-        << " AckInterval=" << m_iACKInterval
-        << " pkt-count=" << m_iPktCount << " liteack-count=" << m_iLightACKCount);
+HLOGC(mglog.Debug, log << CONID() << "checkTimers: nextacktime=" << FormatTime(m_ullNextACKTime_tk)
+    << " AckInterval=" << m_iACKInterval
+    << " pkt-count=" << m_iPktCount << " liteack-count=" << m_iLightACKCount);
 #endif
 
     // Check if it is time to send ACK
-    checkACKTimer(currtime_tk);
+    checkACKTimer(currtime);
 
     // Check if it is time to send a loss report
-    checkNAKTimer(currtime_tk);
+    checkNAKTimer(currtime);
 
     // Check if the connection is expired
-    if (checkExpTimer(currtime_tk))
+    if (checkExpTimer(currtime))
         return;
 
     // Check if FAST or LATE packet retransmission is required
-    checkRexmitTimer(currtime_tk);
+    checkRexmitTimer(currtime);
 
+    const steady_clock::time_point curtp = srt::sync::steady_clock::now();
```

Comment:
Why is it necessary to take check time again here instead of using existing 'currtime'? If intended, please add a short comment explaining this change.